lib/active_job/queue_adapters/pubsub_adapter.rb

```diff
@@ -7,15 +7,34 @@ class PubsubAdapter
       #
       # @param [ActiveJob::Base] job The job to be performed.
       def enqueue(job)
-        raise(NotImplementedError)
+        ActiveSupport::Notifications.instrument("enqueue.pubsub_adapter") do
+          # puts("\n [PubsubAdapter][enqueue]: #{job.inspect}")
+          publish_job(job)
+        end
       end
 
       # Enqueue a job to be performed at a certain time.
       #
       # @param [ActiveJob::Base] job The job to be performed.
       # @param [Float] timestamp The time to perform the job.
       def enqueue_at(job, timestamp)
-        raise(NotImplementedError)
+        ActiveSupport::Notifications.instrument("enqueue_at.pubsub_adapter") do
```

Comment:
Same here: https://guides.rubyonrails.org/active_support_instrumentation.html#active-job

Reply:
Included on commit: https://github.com/skysemilla/kisi-api-challenge/commit/9a21d53a0541cc08008bc1441cee7f3ef974f5bd. I removed the `instrument` and adjusted the parameters on `subscribe` to listen to accurate events.